### DISCUSSION TITLE
What is the best way to open a Sheet from Data Table actions

### DISCUSSION THREAD

**Discussion by JamesSingleton** [shadcn-ui/ui#3319] [Q&A]
I am trying to figure out the best way to open the [Sheet](https://ui.shadcn.com/docs/components/sheet) component from the [Data Table](https://ui.shadcn.com/docs/components/data-table). My current solution has a `<Sheet />` per row which probably isn't ideal.

Here is the [data table](https://github.com/JamesSingleton/project-aqua/blob/main/apps/admin/components/roster/data-table.tsx) and the [columns](https://github.com/JamesSingleton/project-aqua/blob/main/apps/admin/components/roster/columns.tsx) which I followed from the documentation. However, I want to have a `<Sheet />` open with the data from the row. I feel like my current [solution](https://github.com/JamesSingleton/project-aqua/blob/main/apps/admin/components/roster/columns.tsx#L89-L277) causes too many Sheet components 😅

**Comment by ArielMejiaDev** [comment]
probably adding a constant as a boolean to open and close the sheet and another constant to set the currentValue and every click would set the new value and set as true the constant toggle the sheet would do the trick
